pkg/workloadslicing/workloadslicing.go

```diff
@@ -121,6 +122,13 @@ func FindNotFinishedWorkloads(ctx context.Context, clnt client.Client, jobObject
 
 	// Sort workloads by creation timestamp - oldest first.
 	sort.Slice(list.Items, func(i, j int) bool {
+		// Handle cases when API server timestamps have only second-level precision (integration tests),
+		// which makes it possible for two objects to have equal timestamps.
+		if list.Items[i].CreationTimestamp.Equal(&list.Items[j].CreationTimestamp) {
+			rvI, _ := strconv.ParseInt(list.Items[i].ResourceVersion, 10, 64)
```

Comment:
I tested locally and I can confirm this fixes the issue with tests, but I'm skeptical if this way of fix is intended, because k8s documentation explicitly warns against assuming that ResourceVersion is numeric, see https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions says: **You must not assume resource versions are numeric or collatable.**.  

IIRC on some clouds under some configurations this is not true.

Reply:
PTAL: https://github.com/kubernetes-sigs/kueue/pull/6203